charon/src/bin/generate-ml/main.rs

```diff
@@ -1369,6 +1392,13 @@ fn generate_ml(
                     "PathElem",
                     "Name",
                     "ItemMeta",
+                ]),
+                (GenerationKind::TypeDecl(Some(DeriveVisitors {
+                    name: "type_decl",
+                    ancestor: Some("type_decl_base"),
+                    reduce: false,
+                    extra_types: &[],
```

Comment:
I think this would avoid the need for the manual intermediate visitor:
```suggestion
                    ancestor: Some("generic_params"),
                    reduce: false,
                    extra_types: &["attr_info", "item_meta"]
```

Reply:
Ah yes indeed